libraries/AP_LTM_Telem/AP_LTM_Telem.cpp

```diff
@@ -0,0 +1,233 @@
+    send_LTM(lt_buff, LTM_SFRAME_SIZE);
+    _ltm_scheduler++;
+}
+
+// Attitude frame
+void AP_LTM_Telem::generate_Aframe(void)
+{
+    get_attitude();
+
+    uint8_t lt_buff[LTM_AFRAME_SIZE];
+    // A Frame: $T(2 bytes)A(1byte)PITCH(2 bytes)ROLL(2bytes)HEADING(2bytes)CRC(xor,1byte)
+    // START
+    lt_buff[0] = 0x24; //$
+    lt_buff[1] = 0x54; //T
+    // FRAMEID
+    lt_buff[2] = 0x41; //A 
+    // PAYLOAD
+    lt_buff[3] = (_uav.pitch >> 8 * 0) & 0xFF;
+    lt_buff[4] = (_uav.pitch >> 8 * 1) & 0xFF;
+    lt_buff[5] = (_uav.roll >> 8 * 0) & 0xFF;
+    lt_buff[6] = (_uav.roll >> 8 * 1) & 0xFF;
+    lt_buff[7] = (_uav.heading >> 8 * 0) & 0xFF;
+    lt_buff[8] = (_uav.heading >> 8 * 1) & 0xFF;
+    send_LTM(lt_buff, LTM_AFRAME_SIZE);
+    _ltm_scheduler++;
+}
+
+// Send LTM
+void AP_LTM_Telem::generate_LTM(void)
+{
+    if (_ltm_scheduler & 1) {   // is odd
+        generate_Aframe();
+    } else {                    // is even
+        if (_ltm_scheduler % 4 == 0) {
+            generate_Sframe();
+        } else {
+            generate_Gframe();
+        }
+    }
+    if (_ltm_scheduler > 9) {
+        _ltm_scheduler = 1;
+    }
+}
+
+void AP_LTM_Telem::get_position(void)
+{
+    const AP_GPS &gps = AP::gps();
+    const AP_AHRS &ahrs = AP::ahrs();
+
+    Location loc;
+
+    float alt;
+    ahrs.get_relative_position_D_home(alt);
+
+    if (ahrs.get_position(loc)) {
+        _uav.lat = loc.lat;
+        _uav.lon = loc.lng;
+        _uav.gndspeed = (uint8_t) roundf(gps.ground_speed());
+        _uav.alt = (int32_t) roundf(alt);
+        _uav.sats_visible = (uint8_t) AP::gps().num_sats();
+        _uav.fix_type = (uint8_t) AP::gps().status();
+    } else {
+        _uav.lat = 0;
+        _uav.lon = 0;
+        _uav.gndspeed = 0;
+        _uav.alt = (int32_t) roundf(alt);
+        _uav.sats_visible = (uint8_t) AP::gps().num_sats();
+        _uav.fix_type = (uint8_t) AP::gps().status();
+    }
+}
+
+void AP_LTM_Telem::get_sense(void)
+{
+    const AP_BattMonitor &battery = AP::battery();
+    const AP_AHRS &ahrs = AP::ahrs();
+    const AP_Airspeed *aspeed = ahrs.get_airspeed();
+    AP_RSSI *ap_rssi = AP_RSSI::get_singleton();
+
+    _uav.volt = (uint16_t) roundf(battery.voltage() * 10.0f);
+    // _uav.amp = (uint16_t) roundf(battery.current_amps() * 10.0f);
+    float current;
+    if (!battery.current_amps(current)) {
+        current = 0;
+    }
+    _uav.amp = (uint16_t) roundf(current * 10.0f);
+
+    // airspeed in m/s if available and enabled - even if not used - otherwise send 0
+    if (aspeed && aspeed->enabled()) {
+        _uav.airspeed = (uint8_t) roundf(aspeed->get_airspeed() / 10.0f);
+    } else {
+        _uav.airspeed = 0;
+    }
+
+    _uav.flightmode = AP_Notify::flags.flight_mode;
+
+    if (ap_rssi) {
+        uint8_t rssiv = ap_rssi->read_receiver_rssi_uint8();
+        _uav.rssi = (rssiv * 99) / 255;
+    } else

```

Comment:
style requires enclosing braces around single expressions.

Reply:
actually I'm not sure what should get enclosed - _(rssiv * 99) / 255_ expression to _((rssiv * 99) / 255)_? btw, this code was copied from https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_OSD/AP_OSD_Screen.cpp#L926-L928

Reply:
Fixed - I didn't notice that brace missing on last line before.